### PR TITLE
Update read me about samples

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-c/issues/1283

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,9 +193,9 @@ The following CMake options are available for adding/removing project features.
 </tr>
 </table>
 
-- ``Samples``: Whenever UNIT_TESTING is ON, samples are built using the default PAL (see [running samples section](#running-samples)). This means that running samples would throw errors like:
+- ``Samples``: Storage Samples are built by default using the default PAL and HTTP adapter (see [running samples](#running-samples)). This means that running samples without building an HTTP transport adapter would throw errors like:
```
Comment (ahsonkhan) [inline README.md:196]: We should have a `BUILD_SAMPLES` option to simplify this.

File: README.md [hunk 2]
```diff
@@ -240,9 +240,12 @@ The following compilation, preprocessor options will add or remove functionality
 
 ## Running Samples
 
-See [cmake options](#cmake-options) to learn about how to build samples with HTTP implementation in order to be runnable.
+See [cmake options](#cmake-options) to learn about how to build an HTTP transport adapter and how to set PAHO cmake option to build IoT samples.
```
Comment (ahsonkhan) [inline README.md:243]:
```suggestion
See [cmake options](#cmake-options) to learn about how to build an HTTP transport adapter, how to build IoT samples, and to turn logging on.

```

File: README.md [hunk 3]
```diff
@@ -240,9 +240,12 @@ The following compilation, preprocessor options will add or remove functionality
 
 ## Running Samples
 
-See [cmake options](#cmake-options) to learn about how to build samples with HTTP implementation in order to be runnable.
+See [cmake options](#cmake-options) to learn about how to build an HTTP transport adapter and how to set PAHO cmake option to build IoT samples.
 
-After building samples with HTTP stack, set the environment variables for credentials. The samples read these environment values to authenticate to Azure services. See [client secret here](https://docs.microsoft.com/azure/active-directory/azuread-dev/v1-oauth2-on-behalf-of-flow#service-to-service-access-token-request) for additional details on Azure authentication.
+### Storage Sample
+Storage sample would expect storage account with container and SaS to be set in an environment variable `AZURE_STORAGE_URL`.
```
Comment (ahsonkhan) [inline README.md:246]:
```suggestion
The storage sample expects a storage account with a container and SaS token used for authentication to be set in an environment variable `AZURE_STORAGE_URL`.
```

File: README.md [hunk 4]
```diff
@@ -265,6 +268,11 @@ The reason for this is the fact of this functions are not thread-safe, and a cus
 
 **This is libcurl specific only.**
 
+### IoT samples
+Samples for IoT will be built only when CMake option `TRANSPORT_PAHO` is set.
+See [compiler options](#compiler-options).
+For more information about IoT project and samples see [IoT](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/docs/iot).
```
Comment (ahsonkhan) [inline README.md:274]:
```suggestion
For more information about IoT APIs and samples, see [IoT](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/docs/iot).

```

**Review by ahsonkhan — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ahsonkhan) [reply README.md:196]: Oh, we do `AZ_SDK_C_NO_SAMPLES`.

Are storage samples being built even when `AZ_SDK_C_NO_SAMPLES` is on?

Comment (vhvb1989) [reply README.md:196]: `AZ_SDK_C_NO_SAMPLES` disables all samples generation.

**Pull request merged**